components/script/dom/webgl2renderingcontext.rs

```diff
@@ -184,7 +212,22 @@ impl WebGL2RenderingContextMethods for WebGL2RenderingContext {
 
     /// https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.5
     fn BindBuffer(&self, target: u32, buffer: Option<&WebGLBuffer>) {
-        self.base.BindBuffer(target, buffer)
+        let slot = match target {
+            constants::UNIFORM_BUFFER => &self.bound_uniform_buffer,
+            constants::TRANSFORM_FEEDBACK_BUFFER => &self.bound_transform_feedback_buffer,
+            _ => return self.base.BindBuffer(target, buffer),
+        };
+
+        if let Some(buffer) = buffer {
```

Comment:
I think you still want to call down to self.base.BindBuffer() here -- you've lost the validation of the enum to be either array or element_array.